File: src/main/java/io/github/marcuscastelo/invtweaks/InvTweaksOperationInfo.java
package io.github.marcuscastelo.invtweaks;

import io.github.marcuscastelo.invtweaks.inventory.ScreenInventory;
import net.minecraft.screen.slot.Slot;

public class InvTweaksOperationInfo {
    public final InvTweaksOperationType type;

    public final Slot clickedSlot;

    public final ScreenInventory clickedInventoryBoundInfo;

    public final ScreenInventory otherInventoryBoundInfo;

    public InvTweaksOperationInfo(InvTweaksOperationType type, Slot clickedSlot, ScreenInventory clickedInventoryBoundInfo, ScreenInventory otherInventoryBoundInfo) {
        this.type = type;
        this.clickedSlot = clickedSlot;
        this.clickedInventoryBoundInfo = clickedInventoryBoundInfo;
        this.otherInventoryBoundInfo = otherInventoryBoundInfo;
    }

    public InvTweaksOperationInfo(InvTweaksOperationType type, Slot clickedSlot, ScreenInventory clickedInventory) {
        this(type, clickedSlot, clickedInventory, null);
    }
}

File: src/main/kotlin/io/github/marcuscastelo/invtweaks/InvTweaksOperationInfo.kt
package io.github.marcuscastelo.invtweaks

import io.github.marcuscastelo.invtweaks.inventory.ScreenInventories
import io.github.marcuscastelo.invtweaks.inventory.ScreenInventory
import net.minecraft.screen.slot.Slot

data class InvTweaksOperationInfo(
        val type: InvTweaksOperationType,
        val clickedSlot: Slot,
        val clickedSI: ScreenInventory,
        val targetSI: ScreenInventory,
        val otherInventories: ScreenInventories)
{
    //TODO: refactor those accessors
    fun type() = type
    fun clickedSlot() = clickedSlot
    fun clickedSI() = clickedSI
    fun targetSI() = targetSI
    fun otherInventories() = otherInventories
}
